src/test_suite/multiprocessing_utils.py

```diff
@@ -314,11 +314,8 @@ def build_test_results(
             effects = harness_ctx.effects_type()
             effects.ParseFromString(result)
 
-            if globals.consensus_mode:
-                harness_ctx.diff_effect_fn = harness_ctx.consensus_diff_effect_fn
-
-            # Note: diff_effect_fn may modify effects in-place
-            all_passed &= harness_ctx.diff_effect_fn(ref_effects, effects)
+            # Note: apply_diff may modify effects in-place
+            all_passed &= globals.diff_mode.apply_diff(ref_effects, effects)
```

Comment:
This will break functionality for all of our other harnesses (e.g. txn, syscall, etc) since they don't implement an `apply_diff` function. Can you either keep the core BPF diffing logic within the consensus diff function, or create a new field in `HarnessCtx` (see `fuzz_interface.py` and `fuzz_context.py`) that allows harnesses to specify their own consensus diff function? e.g. InstrHarness would look like this:
```
InstrHarness = HarnessCtx(
    fuzz_fn_name="sol_compat_instr_execute_v1",
    fixture_desc=invoke_pb.InstrFixture.DESCRIPTOR,
    prune_effects_fn=instr_prune.prune_execution_result,
    context_human_encode_fn=instr_codec.encode_input,
    context_human_decode_fn=instr_codec.decode_input,
    effects_human_encode_fn=instr_codec.encode_output,
    consensus_diff_effect_fn=instr_diff.consensus_instr_diff_effects,
    core_bpf_diff_effects_fn = instr_diff.core_bpf_instr_diff_effects
)
```

and then you'd change `HarnessCtx` to include an additional `core_bpf_diff_effects_fn` that points to the generic diff function, like this:
```
@dataclass
class HarnessCtx:
    fuzz_fn_name: str
    fixture_desc: InitVar[descriptor.Descriptor]
    result_field_name: str | None = "result"
    diff_effect_fn: Callable[[EffectsType, EffectsType], bool] = generic_effects_diff
    core_bpf_diff_effects_fn: Callable[[EffectsType, EffectsType], bool] = generic_effects_diff
    ...
```

Reply:
You will also be able to remove [these lines](https://github.com/firedancer-io/solana-conformance/pull/103/files#diff-3d96cde513625609c09f6400b955dea4bbc89862c81a8305853a336bcd686217R5-R21) after implementing this.

Reply:
Gotcha!! No problem. Just made that change.